### PR TITLE
Update Scala maven plugin to compile on java 11

### PR DESCRIPTION
**Pull request by aahmed-se**
This updates the scala compiler plugin and also add java source and target properties to successfully start compilation and packaging with openjdk11.

### CODE REVIEW

**Comment by aahmed-se** [comment]
run java8

**Comment by aahmed-se** [comment]
run java8 tests

File: examples/flink/pom.xml [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
   <properties>
     <log4j2.version>2.10.0</log4j2.version>
+    <maven.compiler.target>1.8</maven.compiler.target>
+	  <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (sijie) [inline examples/flink/pom.xml]: use spaces

File: pulsar-flink/pom.xml [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
     <dependency>
       <groupId>org.apache.flink</groupId>
-      <artifactId>flink-table_${scala.binary.version}</artifactId>
+      <artifactId>flink-table-planner_${scala.binary.version}</artifactId>
```
Comment (sijie) [inline pulsar-flink/pom.xml]: have you verified that the current flink connectors are working well in 1.8.0? If not, I would suggest leaving flink version unchanged for now, until you verify it.

File: pom.xml [hunk 3]
```diff
@@ -185,8 +187,8 @@ flexible messaging model and an intuitive client API.</description>
     <mysql-jdbc.version>8.0.11</mysql-jdbc.version>
     <elasticsearch.version>6.3.2</elasticsearch.version>
     <presto.version>0.206</presto.version>
-    <flink.version>1.6.0</flink.version>
-    <scala.binary.version>2.11</scala.binary.version>
+    <flink.version>1.8.0</flink.version>
+    <scala.binary.version>2.12</scala.binary.version>
```
Comment (sijie) [inline pom.xml]: any particular reason that you are changing scala version from 2.11 to 2.12? Changing scala version can potentially impacting the down stream application who is using flink. I would prefer leaving it as 2.11 for now unless there is a strong reason call out for changing this to 2.12.

Comment (aahmed-se) [reply pom.xml]: ok reverted the change

File: pulsar-flink/pom.xml (hunk 2 above)
Comment (aahmed-se) [reply pulsar-flink/pom.xml]: reverted

File: examples/flink/pom.xml (hunk 1 above)
Comment (aahmed-se) [reply examples/flink/pom.xml]: done

**Comment by merlimat** [comment]
@sijie changed the title. this is now only upgrading the scala-maven-plugin version

**Pull request merged**